### ISSUE TITLE
Auto-completion / Search doesn't work with the command CD

### ISSUE THREAD

**Issue by 0xbaadf00d** [fish-shell/fish-shell#2970]
Hi,

The auto-completion (by using TAB) and search mode doesn't work with the built-in command `cd`.
 
### Reproduction Steps:

1. Open a shell (or a remote SSH connexion)
2. Type "cd /"
3. Use TAB to enable auto-completion.
4. Use TAB a second time  (bug: nothing happen)
5. Type some letters to entering search mode (bug: nothing happen)

### Expected behavior:

With other commands (even with does not exist commands), the second TAB enable a _ncurse_ like graphical item selector OR if we type something, a "search" mode is activated.

### Observed behavior:

[Describe observed behavior here]

### Additional information:

-  __When did this problem start?__ Since FISH was intall
-  __Does this happen all the time?__ Yes
-  __Has anything changed on your system recently?__ No, tested on fresh install
  

---
Fish version: fish, version 2.2.0

Operating system: CentOS 7 amd64

```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

Terminal or terminal emulator: Tested on PuTTY and Xterm (Gnome / CentOS)

![issue-screen](https://cloud.githubusercontent.com/assets/1005086/14766338/bfdcfd46-0a05-11e6-93bc-96b86cebb8d0.png)

**Comment by krader1961** [comment]
In your screen shots you're typing uppercase `CD`. That's why fish is saying `Unknown command 'CD'`.

The double-tab completion behavior is fixed in the 2.3.0 release that just entered beta testing. In the 2.2.0 release you can enter the pager mode to select from the available selections by pressing the arrow keys.

**Comment by 0xbaadf00d** [comment]
The issue is not about bad command, is that just another test case:

- `cd` (in lower case) work but FISH does not give auto-completion neither search mode
- `CD` (in upper case) or ANY other non existing command : FISH give auto-completion and search mode
- `rm` (in lower case) command : FISH give auto-completion and search mode

**Comment by ridiculousfish** [comment]
What does `ls /` show?

**Comment by 0xbaadf00d** [comment]
![tab-tab-fish](https://cloud.githubusercontent.com/assets/1005086/14775214/a35a7b9a-0abd-11e6-8bae-c206fa9e6cc2.png)


`ls` command working with auto-completion and search mode.

**Comment by ridiculousfish** [comment]
Thank you for those nice screenshots. This is definitely odd.

**Comment by ridiculousfish** [comment]
git bisect shows this was fixed in 0a99772572ab8632f103f05570da707683980d17, which was faho's big cd completion rewrite, which is in 2.3 (currently in beta).

If you have time, I'd appreciate your confirmation by testing the beta. You can get the package for CentOS [here](http://software.opensuse.org/download.html?project=shells%3Afish%3Abeta%3A2&package=fish). Thanks if so! (If you don't have the time or inclination, no worries - I was able to verify the fix so I'm comfortable closing).

**Comment by 0xbaadf00d** [comment]
Hi @ridiculousfish ,

Yes it working like a charm with fish 2.3b1 on CentOS 7.

Thanks

**Comment by faho** [comment]
>git bisect shows this was fixed in 0a99772

That one just keeps on giving :smile:.

Since this is confirmed as fixed, we can close it.
